src/librustc_middle/ty/structural_impls.rs

```diff
@@ -842,7 +844,9 @@ impl<'tcx> TypeFoldable<'tcx> for ty::instance::Instance<'tcx> {
         Self {
             substs: self.substs.fold_with(folder),
             def: match self.def {
-                Item(did) => Item(did.fold_with(folder)),
+                // We don't fold the param `DefId` here,
+                // as it should only be used for `type_of`.
```

Comment:
Could you expand on this comment a little, for a reader who may not have the full context?

Reply:
Folding `DefId`s is a noop anyways :laughing: Kind of weird that we explicitly fold them here :thinking: We don't do this while folding `TyKind`